allenact/embodiedai/models/visual_nav_models.py

```diff
@@ -0,0 +1,210 @@
+from typing import Callable, Tuple, Dict, Optional, Union, List, cast, Any
+from allenact.utils.system import get_logger
+from collections import OrderedDict
+
+import gym
+import torch
+import torch.nn as nn
+from gym.spaces.dict import Dict as SpaceDict
+
+from allenact.algorithms.onpolicy_sync.policy import (
+    ActorCriticModel,
+    LinearCriticHead,
+    LinearActorHead,
+    ObservationType,
+    DistributionType,
+)
+from allenact.base_abstractions.distributions import CategoricalDistr
+from allenact.base_abstractions.misc import ActorCriticOutput, Memory
+from allenact.utils.model_utils import FeatureEmbedding
+from allenact.embodiedai.models.basic_models import RNNStateEncoder
+from allenact.embodiedai.models.aux_models import AuxiliaryModel
+from allenact.embodiedai.models.fusion_models import FusionModels
+from allenact.embodiedai.aux_losses.losses import TaskWeightsLoss
+
+
+class VisualNavActorCritic(ActorCriticModel[CategoricalDistr]):
+    def __init__(
+        self,
+        action_space: gym.spaces.Discrete,
+        observation_space: SpaceDict,
+        hidden_size=512,
+        multiple_beliefs=False,
+        beliefs_fusion: Optional[str] = None,
+        auxiliary_uuids: Optional[List[str]] = None,
+    ):
+        super().__init__(action_space=action_space, observation_space=observation_space)
```

Comment:
I'd add an `assert multiple_beliefs == beliefs_fusion is not None` here so that people don't accidentally think they're using multiple beliefs simply by specifying a fusion method.